liquibase-bigquery/src/main/java/liquibase/snapshot/jvm/BigQueryPrimaryKeySnapshotGenerator.java

```diff
@@ -0,0 +1,109 @@
+package liquibase.snapshot.jvm;
+
+import liquibase.CatalogAndSchema;
+import liquibase.Scope;
+import liquibase.database.BigQueryDatabase;
+import liquibase.database.Database;
+import liquibase.exception.DatabaseException;
+import liquibase.executor.Executor;
+import liquibase.executor.ExecutorService;
+import liquibase.snapshot.DatabaseSnapshot;
+import liquibase.snapshot.InvalidExampleException;
+import liquibase.snapshot.SnapshotGenerator;
+import liquibase.statement.core.RawParameterizedSqlStatement;
+import liquibase.structure.DatabaseObject;
+import liquibase.structure.core.Column;
+import liquibase.structure.core.Index;
+import liquibase.structure.core.PrimaryKey;
+import liquibase.structure.core.Schema;
+import liquibase.structure.core.Table;
+
+import java.util.List;
+import java.util.Map;
+
+public class BigQueryPrimaryKeySnapshotGenerator extends PrimaryKeySnapshotGenerator {
+
+    @Override
+    public int getPriority(Class<? extends DatabaseObject> objectType, Database database) {
+        if (!(database instanceof BigQueryDatabase)) {
+            return PRIORITY_NONE;
+        }
+        int priority = super.getPriority(objectType, database);
+        if (priority > PRIORITY_NONE) {
+            priority += PRIORITY_DATABASE;
+        }
+        return priority;
+    }
+
+    @Override
+    public Class<? extends SnapshotGenerator>[] replaces() {
+        return new Class[]{PrimaryKeySnapshotGenerator.class};
+    }
+
+    @Override
+    protected DatabaseObject snapshotObject(DatabaseObject example, DatabaseSnapshot snapshot) throws DatabaseException, InvalidExampleException {
+        Database database = snapshot.getDatabase();
+        Schema schema = example.getSchema();
+        String searchTableName = null;
+        if (((PrimaryKey) example).getTable() != null) {
+            searchTableName = ((PrimaryKey) example).getTable().getName();
+            searchTableName = database.correctObjectName(searchTableName, Table.class);
+        }
+        PrimaryKey returnKey = null;
+
+        String keyColumnUsageStatement = String.format("SELECT * FROM %s.INFORMATION_SCHEMA.KEY_COLUMN_USAGE WHERE table_name = ?", schema.getSchema());
+        Executor executor = Scope.getCurrentScope().getSingleton(ExecutorService.class).getExecutor("jdbc", database);
+        List<Map<String, ?>> maps = executor.queryForList(new RawParameterizedSqlStatement(keyColumnUsageStatement, searchTableName));
+        String columnName;
+        for (Map<String, ?> map : maps) {
+            columnName = map.get("COLUMN_NAME").toString();
```

Comment:
i would replace `obj.toString()` to `Objects.toString(obj,null)` in all places where we get something from map. Never knows when fields are missing and you get NPE